src/sentry/eventstream/kafka/backend.py

```diff
@@ -258,88 +304,89 @@ def handle_shutdown_request(signum: int, frame: Any) -> None:
         signal.signal(signal.SIGINT, handle_shutdown_request)
         signal.signal(signal.SIGTERM, handle_shutdown_request)
 
-        i = 0
-        while not shutdown_requested:
-            message = consumer.poll(0.1)
-            if message is None:
-                continue
+        def run_processing_loop() -> None:
```

Comment:
Why not reusing the batching consumer we already have in the sentry codebase  ?
https://github.com/getsentry/sentry/blob/master/src/sentry/utils/batching_kafka_consumer.py#L95

That code is tested so you would have to only build the process and the flush method.

IT would also make this code simpler.

Reply:
> A change of this size to the post process consumer needs testing coverage.
> Now there is very little end to end testing coverage. IT is really hard to do.
> I think a good strategy to make this change and achieve a better test coverage is reusing the existing batching consumer instead of rebuilding it. Also you could maintain the old implementation and switch when the post process consumer is started.

The primary reason I did not use the batching kafka consumer was that the consumer it creates and uses is a simple consumer which follows one topic. It is not a synchronized consumer which the post process forwarder needs. Having said that, let me take a look at the interfaces of the consumer it uses and see if I could make it work by passing in a synchronized consumer and have it use that.